installation_and_upgrade/ibex_install_utils/tasks/server_tasks.py

```diff
@@ -327,20 +327,43 @@ def _pretty_print(data):
                     print("Couldn't get data from {} because: {}".format(pv, e.message))
                     f.write(e.message)
 
-    @task("Patch ISISDAE for ticket 5164")
-    def patch_isisdae(self):
-        if BaseTasks._get_machine_name() == "NDEMUONFE":
-            for filename in os.listdir(RELEASE_5_5_0_ISISDAE_DIR):
-                if filename.lower() == "isisdae-ioc-01.exe":
-                    continue
-
-                source_path = os.path.join(RELEASE_5_5_0_ISISDAE_DIR, filename)
-                dest_path = os.path.join(EPICS_PATH, "ioc", "master", "ISISDAE", "bin", "windows-x64", filename)
-
-                print("Patching {}".format(filename))
-                if os.path.exists(dest_path):
-                    os.remove(dest_path)
-                shutil.copy2(source_path, dest_path)
-            print("ISISDAE successfully patched")
+    @task("Update the ICP")
+    def update_icp(self, icp_in_labview_modules):
+        register_icp_commands = AdminCommandBuilder()
+
+        if icp_in_labview_modules:
+            config_filepath = os.path.join(LABVIEW_DAE_DIR, "icp_config.xml")
+            root = lxml.etree.parse(config_filepath)
+            try:
+                DAE_type = int(root.xpath("./I32/Name[text() = 'DAEType']/../Val/text()")[0])
+            except Exception as e:
+                print("Failed to find DAE_type ({}), not installing ICP".format(e.message))
+                return
+            # If the ICP is talking to a DAE2 it's DAEType will be 1 or 2, if it's talking to a DAE3 it will be 3 or 4
+            if DAE_type in [1, 2]:
+                DAE_type = 2
+            elif DAE_type in [3, 4]:
+                DAE_type = 3
+            else:
+                print("DAE type {} not recognised, not installing ICP".format(DAE_type))
+                return
+            self.prompt.confirm_step("Upgrade DAE{} type ICP found in Labview Modules".format(DAE_type))
+            ICP_path = get_latest_directory_path(os.path.join(INST_SHARE_AREA, "kits$", "CompGroup", "ICP", "ISISICP",
+                                                              "DAE{}".format(DAE_type)), "")
+
+            RunProcess(os.getcwd(), "update_inst.cmd", executable_directory=ICP_path).run()
```

Comment:
update_inst now always does a batch "pause" at the end unless you pass NOINT  as the first argument.